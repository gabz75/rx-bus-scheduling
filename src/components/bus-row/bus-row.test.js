import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';

import BusRow from './index';

const MOCK_BUS = { id: 1, trips: [] };

it('renders', () => {
  const { asFragment } = render(<BusRow bus={MOCK_BUS} />);
  expect(asFragment()).toMatchSnapshot();
});

it('executes a callback with bus as argument', () => {
  const onClickHandler = jest.fn();
  const { getByTestId } = render(<BusRow bus={MOCK_BUS} onClick={onClickHandler} />);
  getByTestId('bus-row').click();
  expect(onClickHandler).toHaveBeenCalledWith(MOCK_BUS);
});

it('renders with children', () => {
  const { getByTestId } = render(<BusRow bus={MOCK_BUS}><span>test</span></BusRow>);
  expect(getByTestId('bus-row')).toContainHTML('<span>test</span>');
});
