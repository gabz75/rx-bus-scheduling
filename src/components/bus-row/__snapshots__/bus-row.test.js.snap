// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-AykKC gFqVvT"
    data-testid="bus-row"
  />
</DocumentFragment>
`;
